src/meta/src/barrier/rpc.rs

```diff
@@ -196,26 +210,49 @@ impl ControlStreamManager {
                             .expect("should exist when get collect resp");
                         break Ok((worker_id, command.prev_epoch.value().0, resp));
                     }
-                    resp => {
-                        break Err(anyhow!("get unexpected resp: {:?}", resp).into());
+                    Response::Shutdown(resp) => {
+                        let node = self
+                            .nodes
+                            .remove(&worker_id)
+                            .expect("should exist when get shutdown resp");
+                        let has_inflight_barrier = !node.inflight_barriers.is_empty();
```

Comment:
Note that, in https://github.com/risingwavelabs/risingwave/pull/17758, `inflight_barriers` will be removed